game/scripts/vscripts/abilities/stopfightingyourself/dupe_heroes.lua

```diff
@@ -0,0 +1,147 @@
+    self:GetSpecialValueFor('radius'),
+    self:GetAbilityTargetTeam(),
+    self:GetAbilityTargetType(),
+    self:GetAbilityTargetFlags(),
+    FIND_ANY_ORDER,
+    false
+  )
+
+
+  for _,unit in ipairs(units) do
+    if unit:IsRealHero() and unit ~= caster then
+      self:StartCooldown(self:GetSpecialValueFor('cooldown'))
+
+      local illusion = CreateUnitByName(
+        unit:GetName(),
+        unit:GetAbsOrigin(),
+        true,
+        caster,
+        caster,
+        --nil,
+        --nil,
+        caster:GetTeamNumber()
+      )
+
+      -- Level
+      local target_level = unit:GetLevel()
+      for i = 1, target_level - 1 do
+        illusion:HeroLevelUp(false)
+        HeroProgression:ReduceStatGain(illusion, i)
+      end
+
+      illusion:SetAbilityPoints(0)
+
+      -- Abilities
+      for slot = 0, 15 do
+        local theirAbility = unit:GetAbilityByIndex(slot)
+
+        if theirAbility then
+          local ourAbility = illusion:FindAbilityByName(theirAbility:GetAbilityName())
+          if ourAbility then
+            ourAbility:SetLevel(theirAbility:GetLevel())
+          end
+        end
+      end
+
+
+      -- Items
+      for slot = DOTA_ITEM_SLOT_1, DOTA_ITEM_SLOT_6 do
+        local theirItem = unit:GetItemInSlot(slot)
+
+        if theirItem then
+          if not contains(theirItem:GetName(), blacklist) then
+            local ourItem = illusion:AddItemByName(theirItem:GetAbilityName())
+
+            if ourItem:RequiresCharges() then
+              local charges = theirItem:GetCurrentCharges()
+              ourItem:SetCurrentCharges(charges)
+            end
+          end
+        end
+      end
+
+
+      -- Illusion Modifier
+      illusion:AddNewModifier(
+        caster,
+        self,
+        "modifier_illusion",
+        {
+          duration = -1,
+          outgoing_damage = self:GetSpecialValueFor('illusion_outgoing_damage'),
+          incoming_damage = self:GetSpecialValueFor('illusion_incoming_damage'),
+        }
+      )
+
+      --illusion:MakeIllusion()
+
+      -- Stats
+      -- BUG Gold bounty does not work
+      -- NOTE XP probably too
+      illusion:SetMaximumGoldBounty(0)
+      illusion:SetMinimumGoldBounty(0)
+      illusion:SetDeathXP(0)
+      illusion:SetRespawnsDisabled(true)
+
+      -- Set Stats to original hero
+      illusion:SetHealth(unit:GetHealth())
+      illusion:SetMana(unit:GetMana())
+
+      -- Randomly play sound to player (10% chance)
+      -- NOTE this doesn't seem to work
+      if math.random(100) <= 10 then
+        --EmitAnnouncerSoundForPlayer('sounds/vo/announcer_dlc_rick_and_morty/generic_illusion_based_hero_02.vsnd', unit:GetPlayerID())
+        illusion:EmitSound('sounds/vo/announcer_dlc_rick_and_morty/generic_illusion_based_hero_02.vsnd')
+      end
+
+      if caster.illusions == nil then
+        caster.illusions = {}
+      end
+      caster.illusions[illusion:entindex()] = illusion
+
+      illusion:OnDeath(function()
+        -- create particle
```

Comment:
pop a `caster.illusions[illusion:entindex()] = nil` in here.
I think we should also do `cast.illusions.count = cast.illusions.count + 1` when adding in the illusion entry and then `- 1` here... That would allow us to use the count variable in the above conditional to enforce max size.